File: server/tests/fixtures/alpha_rule_triggers/javascript/analysis_change_impact_dependent.js
// Dependent file for analysis.change_impact fixture
// Depends on criticalFunction from analysis_change_impact_core

import { criticalFunction, CriticalService } from './analysis_change_impact_core';

export function useCritical() {
    return criticalFunction("data1");
}

export function useService() {
    const svc = new CriticalService();
    return svc.process("data2");
}

File: server/tests/fixtures/alpha_rule_triggers/python/analysis_change_impact_dependent.py
# Dependent file for analysis.change_impact fixture
# Depends on critical_function from analysis_change_impact

from analysis_change_impact import critical_function, CriticalService

def use_critical():
    """Uses the critical function."""
    return critical_function("data1")

def use_service():
    """Uses the critical service."""
    svc = CriticalService()
    return svc.process("data2")
